### PR TITLE
[RORDEV-655] Ror kibana rule and JWT auth rule with "groups_and" support

### PR DESCRIPTION
**Pull request by mateuszkp96**
🚀New (ES) Added `groups_and` sub-rule for `ror_kbn_auth`, and `jwt_auth`

### CODE REVIEW

File: core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/JwtAuthRule.scala [hunk 1]
```diff
@@ -71,10 +72,10 @@ final class JwtAuthRule(val settings: JwtAuthRule.Settings,
           case RequestGroup.`N/A` =>
             authorizeUsingJwtToken(blockContext)
           case RequestGroup.AGroup(group) =>
-            settings.permittedGroups.toList match {
-              case Nil =>
+            settings.groupsLogic match {
+              case GroupsLogic.NotDefined =>
                 authorizeUsingJwtToken(blockContext)
-              case permittedGroups if permittedGroups.contains(group) =>
+              case GroupsLogic.Defined(strategy) if strategy.availableGroupsFrom(UniqueList.of(group)).isDefined =>
```
Comment (coutoPL) [inline core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/JwtAuthRule.scala:78]: `strategy.groups.contains(group)` we don't need to check here based on strategy (so the `groups` extension method will be needed)

File: core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/RorKbnAuthRule.scala [hunk 2]
```diff
@@ -67,10 +68,10 @@ final class RorKbnAuthRule(val settings: Settings,
         case RequestGroup.`N/A` =>
           authorizeUsingJwtToken(blockContext)
         case RequestGroup.AGroup(group) =>
-          settings.permittedGroups.toList match {
-            case Nil =>
+          settings.groupsLogic match {
+            case GroupsLogic.NotDefined =>
               authorizeUsingJwtToken(blockContext)
-            case permittedGroups if permittedGroups.contains(group) =>
+            case GroupsLogic.Defined(strategy) if strategy.availableGroupsFrom(UniqueList.of(group)).isDefined =>
```
Comment (coutoPL) [inline core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/RorKbnAuthRule.scala:74]: the same here

File: tests-utils/src/main/scala/tech/beshu/ror/utils/httpclient/RestClient.scala [hunk 3]
```diff
@@ -92,7 +92,7 @@ class RestClient(ssl: Boolean,
   private def withSsl(builder: HttpClientBuilder): HttpClientBuilder = {
     val sslCtxBuilder = new SSLContextBuilder();
     sslCtxBuilder.loadTrustMaterial(null, new TrustAllCertificatesStrategy());
-    val sslsf = new SSLConnectionSocketFactory(sslCtxBuilder.build(), ALLOW_ALL_HOSTNAME_VERIFIER);
+    val sslsf = new SSLConnectionSocketFactory(sslCtxBuilder.build(), NoopHostnameVerifier.INSTANCE);
```
Comment (coutoPL) [inline tests-utils/src/main/scala/tech/beshu/ror/utils/httpclient/RestClient.scala:95]: why do we need to change it?

**Review by coutoPL — CHANGES_REQUESTED** [review]
and:
1. please add chanelog style description to the PR desc
2. don't forget about docs

File: tests-utils/src/main/scala/tech/beshu/ror/utils/httpclient/RestClient.scala (hunk 3 above)
Comment (mateuszkp96) [reply tests-utils/src/main/scala/tech/beshu/ror/utils/httpclient/RestClient.scala:95]: I fixed deprecation warning

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**